packages/workflow-apps-defaults/src/index.js

```diff
@@ -3,67 +3,111 @@
+  switch (process.platform) {
+    case 'darwin': {
+      const browserName = (await defaultBrowser()).name.toLowerCase();
+      let Browser;
+      try {
+        Browser = require(`workflow-app-${browserName}`);
+      } catch (error) {
+        console.log(`Cannot find ${browserName}, falling back to safari`);
+        Browser = require('workflow-app-safari');
+      }
+      return {
+        Browser,
+        Terminal: require('workflow-app-iterm'),
+        TextEditor: require('workflow-app-atom'),
+      };
+    }
+    case 'win32':
+      return {
+        Terminal: require('workflow-app-powershell'),
+        Browser: require('workflow-app-chrome'),
+        TextEditor: require('workflow-app-notepad'),
+      };
+    case 'linux': {
+      const browserName = (await defaultBrowser()).name.toLowerCase();
+      let Browser;
+      try {
+        Browser = require(`workflow-app-${browserName}`);
+      } catch (error) {
+        console.log(`Cannot find ${browserName}, falling back to chrome`);
+        Browser = require('workflow-app-chrome');
+      }
+      return {
+        Browser,
+        Terminal: require('workflow-app-xterm'),
+        TextEditor: require('workflow-app-atom'),
+      };
     }
-    throw new Error('not reachable');
-  })();
+    default:
+      console.log(`Platform '${process.platform}' not supported`);
+      console.log(
+        'Look for an issue for your platform here: https://github.com/havardh/workflow/issues'
+      );
+      process.exit(0);
+      break;
+  }
+  throw new Error('not reachable');
+}
 
-  const userDefaults = (() => {
-    try {
-      const path = join(baseFolder, 'apps', 'defaults');
-      const defaultModule = require(path);
+function getUserDefaults() {
+  try {
+    const path = join(baseFolder, 'apps', 'defaults');
+    const defaultModule = require(path);
 
-      return defaultModule.default || defaultModule;
-    } catch (error) {
-      if (error.code !== 'MODULE_NOT_FOUND') {
-        throw error;
-      } else {
-        return {};
-      }
+    return defaultModule.default || defaultModule;
+  } catch (error) {
+    if (error.code !== 'MODULE_NOT_FOUND') {
+      throw error;
+    } else {
+      return {};
     }
-  })();
+  }
+}
 
-  defaultApps.forEach(app => {
-    defaults[app] = userDefaults[app] || platformDefaults[app];
-  });
+let defaults = {};
+async function getDefaultApps() {
+  if (Object.keys(defaults).length > 0) {
+    return defaults;
+  }
+
+  // $FlowSuppress
+  if (process.browser) {
+    defaults = require('workflow-apps-html').defaults;
+  } else {
+    const platformDefaults = await getPlatformDefaults();
+    const userDefaults = getUserDefaults();
+
+    defaultApps.forEach(app => {
+      defaults[app] = userDefaults[app] || platformDefaults[app];
+    });
+  }
+  return defaults;
 }
 
+const Browser = {
+  type: 'async',
+  loader: () => getDefaultApps().then(({ Browser }) => Browser),
+};
+
+const TextEditor = {
```

Comment:
Lets not make this `async` before we need to. :)

Reply:
So that means you agree that we should do https://github.com/havardh/workflow/pull/141#discussion_r216450279 ?